sdk/identity/identity/src/credentials/managedIdentityCredential/appServiceMsi2017.ts

```diff
@@ -25,21 +28,26 @@ function prepareRequestOptions(resource: string, clientId?: string): RequestPrep
     queryParameters.clientid = clientId;
   }
 
+  const query = qs.stringify(queryParameters);
```

Comment:
Do we need to depend on `qs`? 

Can't we do something like:

```ts
const params = new URLSearchParams(queryParameters);
const query = params.toString();
```

ref: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
Agreed. We also had an issue to specifically move away from qs. https://github.com/Azure/azure-sdk-for-js/issues/8841

Reply:
@xirzec On the other hand.... regarding `qs`, given that the identity package already had a dependency on it and it was not introduced in this PR, can the move away from that be done separately from this PR and release?

Or do we believe that we are introducing new usage of `qs` and that increases the risk factor somehow.

Reply:
qs should be fine, I think. I’ll make an issue to drop the dependency on v2

Reply:
I have re-opened https://github.com/Azure/azure-sdk-for-js/issues/8841 to track the qs related work in Identity. Resolving this thread.

Reply:
Somehow I’m still not used in the question mark witchery 😝